pkg/ebpf/tracer.go

```diff
@@ -242,6 +244,86 @@ func (m *FlowFetcher) AttachTCX(iface ifaces.Interface) error {
 	return nil
 }
 
+func removeTCFilters(ifName string, tcDir uint32) error {
+	link, err := netlink.LinkByName(ifName)
+	if err != nil {
+		return err
+	}
+
+	filters, err := netlink.FilterList(link, tcDir)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range filters {
+		if err := netlink.FilterDel(f); err != nil {
```

Comment:
not sure how likely it is to get an error here, but to be more resilient you could catch all errors in a slice and use [errors.Join](https://pkg.go.dev/errors#Join) : that would avoid returning after an error in case there's still filters to delete in the list.

Reply:
I can use Aggregrate error too which does the same like errors join